modules/swagger-codegen/src/main/resources/swift3/Models.mustache

```diff
@@ -202,6 +205,20 @@ class Decoders {
             }{{/isEnum}}
             {{^isEnum}}instance.{{name}} = Decoders.decodeOptional(clazz: {{{baseType}}}.self, source: sourceDictionary["{{baseName}}"] as AnyObject?){{/isEnum}}{{/allVars}}
             {{/unwrapRequired}}
+
+            {{#parent}}
+            for key in keys {
+                propsDictionary.removeValue(forKey: key)
+            }
+
+            for key in propsDictionary.keys {
+                if let decodedValue = Decoders.decodeOptional(clazz: String.self, source: propsDictionary[key] as AnyObject?) {
+                    instance[key as! String] = decodedValue
+                } else {
+                    instance[key as! String] = nil
```

Comment:
Hey @jgavris im happy to change it if you like. But.. 

The properies array is constructed from properies we already know that we have.  The Keys dictionary is a subset of the propertiesDictonary, there will never be a case where we have a key that is nil.    I think we can agree that the key is always going to be of base type string as its a dictonary.  However if your worried about nils.  A nil check here wont be the correct solution as the keys array is an array of none-optional keys.  Ref. line 188

`let keys : [String] = [{{#allVars}}{{^-last}}"{{baseName}}", {{/-last}}{{#-last}}"{{baseName}}"{{/-last}}{{/allVars}}]`

If your worried about nil's we should change this to be an optional string array.  

Again all this I think is unnecessary. 

However its your call.  Let me know.